modules/programs/fish.nix

```diff
@@ -0,0 +1,124 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.fish;
+
+  abbrsStr = concatStringsSep "\n" (
+    mapAttrsToList (k: v: "abbr --add ${k} '${v}'") cfg.shellAbbrs
+  );
+
+  envVarsStr = config.lib.shell.exportAll cfg.sessionVariables;
```

Comment:
This doesn't seem to be actually used anywhere.

Btw, my understanding is that fish doesn't support a posix-like `export` command but perhaps it is still possible to get them in somehow? In particular, it would be nice for the session variables set through `home.sessionVariables`. I found [bass](https://github.com/edc/bass), which might be a way? Perhaps this won't be needed for a first version of the module but might be worth considering for the future?